submodules/users/users.js

```diff
@@ -3682,45 +3689,37 @@ define(function(require) {
 				callback && callback(template, user);
 			});
 		},
-		usersFormatLicensedRolesData: function(user) {
-			var self = this,
-				formattedData = {
-					selectedRole: undefined,
-					availableRoles: self.appFlags.global.servicePlansRole
-				};
 
-			if (user.hasOwnProperty('service') && user.service.hasOwnProperty('plans') && _.size(user.service.plans) > 0) {
-				for (var key in user.service.plans) {
-					if (user.service.plans.hasOwnProperty(key)) {
-						formattedData.selectedRole = key;
-						break;
-					}
-				}
-			}
-
-			return formattedData;
-		},
-		usersFormatDevicesData: function(userId, data) {
+		usersFormatLicensedRolesData: function(user) {
 			var self = this,
-				formattedData = {
-					countSpare: 0,
-					assignedDevices: {},
-					unassignedDevices: {}
-				};
+				userPlanIds = _
+					.chain(user)
+					.get('service.plans', {})
+					.keys()
+					.value();
 
-			_.each(data, function(device) {
-				if (device.owner_id === userId) {
-					formattedData.assignedDevices[device.id] = device;
-				} else if (device.owner_id === '' || !('owner_id' in device)) {
-					formattedData.countSpare++;
-					formattedData.unassignedDevices[device.id] = device;
-				}
+			return _.merge({
+				availableRoles: self.appFlags.global.servicePlansRole
+			}, !_.isEmpty(userPlanIds) && {
+				selectedRole: _.head(userPlanIds)
 			});
+		},
 
-			formattedData.emptyAssigned = _.isEmpty(formattedData.assignedDevices);
-			formattedData.emptySpare = _.isEmpty(formattedData.unassignedDevices);
+		usersFormatDevicesData: function(userId, devices) {
+			var self = this,
+				assigned = _.filter(devices, { owner_id: userId }),
+				unassigned = _.flatten([
+					_.filter(devices, { owner_id: '' }),
+					_.reject(devices, _.partial(_.has, _, 'owner_id'))
+				]);
 
-			return formattedData;
+			return {
+				countSpare: _.size(unassigned),
+				emptyAssigned: _.isEmpty(assigned),
+				emptySpare: _.isEmpty(unassigned),
+				assigned: _.keyBy(assigned, 'id'),
+				unassignedvices: _.keyBy(unassigned, 'id')
```

Comment:
There are some small typos here, based on the original names of the properties:
```suggestion
				assignedDevices _.keyBy(assigned, 'id'),
				unassignedDevices: _.keyBy(unassigned, 'id')
```

Reply:
Thanks, looks like search-and-replace gone wrong.